lib/react_client/react_interop.dart

```diff
@@ -59,16 +56,16 @@ abstract class React {
   external static ReactClass get Fragment;
 
   external static List<dynamic> useState(dynamic value);
-  external static void useEffect(dynamic Function() sideEffect, [List<Object> dependencies]);
-  external static List<dynamic> useReducer(Function reducer, dynamic initialState, [Function init]);
-  external static Function useCallback(Function callback, List dependencies);
+  external static void useEffect(dynamic Function() sideEffect, [List<Object?>? dependencies]);
+  external static List<dynamic> useReducer(Function reducer, dynamic initialState, [Function? init]);
+  external static Function useCallback(Function callback, List<Object?> dependencies);
   external static ReactContext useContext(ReactContext context);
-  external static JsRef useRef([dynamic initialValue]);
-  external static dynamic useMemo(dynamic Function() createFunction, [List<dynamic> dependencies]);
-  external static void useLayoutEffect(dynamic Function() sideEffect, [List<Object> dependencies]);
-  external static void useImperativeHandle(dynamic ref, dynamic Function() createHandle, [List<dynamic> dependencies]);
+  external static JsRef useRef([Object? initialValue]);
+  external static dynamic useMemo(dynamic Function() createFunction, [List<Object?>? dependencies]);
+  external static void useLayoutEffect(dynamic Function() sideEffect, [List<Object?>? dependencies]);
+  external static void useImperativeHandle(dynamic ref, dynamic Function() createHandle, [List<Object?>? dependencies]);
   // NOTE: The use of generics on the `useDebugValue` interop will break the hook.
-  external static dynamic useDebugValue(dynamic value, [Function format]);
+  external static dynamic useDebugValue(dynamic value, [Function? format]);
```

Comment:
Do we know the arity of this function?

Reply:
Yeah, it looks like without generics though (which apparently breaks things; see comment on the previous line), we'd have to type it as `Function(Never)? format` to get the typing correct, which is pretty gross, so I'm inclined to leave it as-is.